### PR TITLE
Flips the bird

### LINKED ISSUE #17845: Someone taught poly a href exploit
The following runtime has occurred 30246 time(s).
runtime error: -censored (Attempted exploit, string output)
proc name: handle automated speech (/mob/living/simple_animal/parrot/handle_automated_speech)
  source file: parrot.dm,380
  usr: null
  src: Poly (/mob/living/simple_animal/parrot/Poly)
  src.loc: the floor (119,80,1) (/turf/open/floor/plasteel)

Everything poly learns runs through html_decode before being learned, so I'm a little baffled what's triggering this

### PR DESCRIPTION
**Pull request by Incoming5643**
Some badmin edited poly's memories in such a way as to break his save in a way that runtimed constantly. This should catch this badmin only problem.

Fixes #17845

### CODE REVIEW

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 1]
```diff
@@ -923,7 +923,7 @@
 	S["longestsurvival"]	>> longest_survival
 	S["longestdeathstreak"] >> longest_deathstreak
 
-	if(isnull(speech_buffer))
+	if(!islist(speech_buffer))
 		speech_buffer = list()
 
 /mob/living/simple_animal/parrot/Poly/proc/Write_Memory()
```
Comment (MrStonedOne) [inline code/modules/mob/living/simple_animal/parrot.dm]: You also need to make it not save if its been fucked with, what for not fucking up the rest of the saved data.

Comment (Incoming5643) [inline code/modules/mob/living/simple_animal/parrot.dm]: Fair enough

**Comment by KorPhaeron** [comment]
>fun feature which learns player phrases and repeats them across rounds
>admins break it inserting their own unfunny joke

Is nothing sacred?

**Comment by Razharas** [comment]
Remove admin varedit

**Comment by callanrocks** [comment]
Remove admins 2016, Callans for headmin!

**Pull request merged**